Doc/library/typing.rst

```diff
@@ -1164,8 +1175,11 @@ These can be used as types in annotations using ``[]``, each having a unique syn
 
 .. data:: Final
 
-   A special typing construct to indicate to type checkers that a name
-   cannot be re-assigned or overridden in a subclass. For example::
+   Special typing construct to indicate final names to type checkers.
+
+   A final name cannot be re-assigned or overridden in a subclass.
```

Comment:
Not just in subclasses, it also applies to globals (as the first example below indicates). Maybe we should have an explicit list of where `Final` can be used.

Reply:
I started writing out an in-depth specification, but quickly realised I'd have to copy over the whole of PEP-591 into the typing docs in order to have everything covered.

We actually _should_ copy over the whole of PEP-591 into the typing docs, but it doesn't belong in the API reference. I have some plans in that regard which will be coming soon, so I'll leave it to another PR. For now, I'll correct the inaccuracy, but will leave the docs for ``Final`` somewhat terse.